src/anyio/from_thread.py

```diff
@@ -486,7 +498,7 @@ def run_blocking_portal() -> None:
                     future.set_exception(exc)
 
     future: Future[BlockingPortal] = Future()
-    thread = Thread(target=run_blocking_portal, daemon=True)
+    thread = NamedThread(target=run_blocking_portal, daemon=True, name=name)
```

Comment:
The idea was to use the ID of the blocking portal, not the thread. The thread ID is already evident from the list of threads.

Reply:
You mean the ID as `id(blocking_portal)`?

Reply:
Yes (presented in the hexadecimal format) as `repr()` does for objects not having their own `__repr__()` method.

Reply:
But we don't have the instance of blocking_portal before starting the thread?

Reply:
That's why we're setting the thread name after the portal instance is created.

Reply:
Oh you set it later when the thread boots